.github/scripts/Dockerfile

```diff
@@ -0,0 +1,54 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM fedora:34 as builder
+
+# FROM quay.io/centos/centos:stream8 as builder
+# Enable additional package repositories for CentOS
+#  note: PowerTools is called CodeReady Linux Builder in RHEL 8
+# RUN dnf -y install epel-release && dnf config-manager --set-enabled powertools && dnf -y install gcc gcc-c++ make cmake cyrus-sasl-devel cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 openssl-devel libuuid-devel nodejs swig wget patch findutils git valgrind libwebsockets-devel python3-devel libnghttp2-devel && dnf clean all -y
+RUN dnf -y install gcc gcc-c++ make cmake cyrus-sasl-devel cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 openssl-devel libuuid-devel nodejs swig wget patch findutils git valgrind libwebsockets-devel python3-devel libnghttp2-devel && dnf clean all -y
+WORKDIR /build
+COPY . .
+ENV PROTON_VERSION=0.36.0
+ENV PROTON_SOURCE_URL=${PROTON_SOURCE_URL:-http://archive.apache.org/dist/qpid/proton/${PROTON_VERSION}/qpid-proton-${PROTON_VERSION}.tar.gz}
+RUN .github/scripts/compile.sh
+
+
+FROM fedora:34
+
+# FROM quay.io/centos/centos:stream8
+# RUN dnf -y install epel-release && dnf config-manager --set-enabled powertools && dnf -y install glibc cyrus-sasl-lib cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 libuuid openssl gettext hostname iputils libwebsockets-devel libnghttp2 gdb valgrind && dnf -y update && dnf clean all
+
+RUN dnf -y install glibc cyrus-sasl-lib cyrus-sasl-plain cyrus-sasl-gssapi cyrus-sasl-md5 libuuid openssl gettext hostname iputils libwebsockets-devel libnghttp2 gdb valgrind && dnf -y update && dnf clean all
```

Comment:
valgrind? I understand that the other packages are there to simplify debugging (gdb), but valgrind?

libwebsockets-devel? Why install a devel package?

cyrus-sasl-md5? Insecure! Burn it, cleanse it with fire!

Reply:
Valgrind is actually needed. Skupper has an initialization option to debug using valgrind or gdb, see:
```
--router-debug-mode string             Enable debug mode for router ('valgrind' or 'gdb' are valid values)
```

There was a reason to keep libwebsockets-devel in the final image, but I do not recall now (I will retest without it).
@ganeshmurthy if you remember why, let me know. Otherwise I will remove and validate if it is really needed.

@grs @ganeshmurthy can we get rid of cyrus-sasl-md5? Any comment?

Reply:
> `--router-debug-mode string`

Whoa! That IMO deserves an explanatory comment in the Dockerfile. Actually, the deps list is somewhat opaque: something is there for proton, something for router, something for skupper's way of dealing with the router.

Reply:
I believe cyrus-sasl-md5 can be removed.